internal/generate/generate.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 The Operator-SDK Authors
```

Comment:
I see that these files are added as new. However, this implementation is not existent? 
So, shows that the project will kept both. IMO would be better here not keep the code duplicated in the repo and if required to do here already your idea for follow up indeed to make clear the motivations of the change.

Reply:
I didn't want to include both changes in a single PR as the cmd bit is going to require some refactoring as well. I wanted to get peoples opinion on the structure of the new interface so I could change it now and not have to redo the cmd part.

Reply:
I would rather see these changes in cmd in this PR so we can comment on whether they make sense from the caller's perspective.

Reply:
HI @jberkhahn,

The problem is how we can ` I wanted to get peoples opinion on the structure of the new interface` if we do not see what was changed/removed? Because of this, IMO it is hard to get your idea from here.

Reply:
Ok, I can include the changes to cmd to show what it'll look like all together.